scripts/azure-price-fetch/azureClientWrapper.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/Azure/go-autorest/autorest/to"
+	"gomodules.xyz/azure-retail-prices-sdk-for-go/sdk"
+)
+
+type AzureClientWrapper struct {
+	subscriptionId string
+	priceClient    *sdk.RetailPricesClient
+	rgClient       *armresources.ResourceGroupsClient
+	vmssClient     *armcompute.VirtualMachineScaleSetsClient
+	vmssVmClient   *armcompute.VirtualMachineScaleSetVMsClient
+}
+
+func NewAzureClientWrapper(subId string, cred *azidentity.DefaultAzureCredential) (*AzureClientWrapper, error) {
+	retailPricesClient, err := sdk.NewRetailPricesClient(nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create retail prices client: %w", err)
+	}
+
+	rgClient, err := armresources.NewResourceGroupsClient(subId, cred, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build resource group client: %w", err)
+	}
+
+	computeClientFactory, err := armcompute.NewClientFactory(subId, cred, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build compute client: %w", err)
+	}
```

Comment:
Very minor, but would likely get flagged once this is moved over to the `pkg` module, consider creating error objects for each scenario(or one for all three?) to make it easier to test each scenario. IE

```
var ErrClientCreationFailure = errors.New("failed to create client")
...
return nil, fmt.Printf("w: %w", ErrClientCreationFailure, err)
```

Reply:
added